### PR TITLE
2398 documentation clarify eventkeep source vars = null in derive extreme event

### PR DESCRIPTION
**Pull request by ProfessorP-beep**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_extreme_event.R [hunk 1]
```diff
@@ -97,6 +97,8 @@
 #'       1. Only the variables specified for the `keep_source_vars` field of the
 #'       event, and the by variables (`by_vars`) and the variables created by
 #'       `set_values_to` are kept.
+#'       1. If `keep_source_vars = NULL` is used for an event in `derive_extreme_event()`
+#'       the value of the `keep_source_vars` argument of `derive_extreme_event()` is used.
```
Comment (bundfussr) [inline R/derive_extreme_event.R:101]: I would not add it as separate item but add it to the previous one:
```suggestion
#'       If `keep_source_vars = NULL` is used for an event in `derive_extreme_event()`,
#'       the value of the `keep_source_vars` argument of `derive_extreme_event()` is used.
```

**Review by bundfussr — COMMENTED** [review]
Could you add to the description of the `keep_source_vars` argument (in `derive_extreme_event()`) that the `keep_source_vars` field of the event takes precedence over the value of the `keep_source_vars` argument?

File: R/derive_extreme_event.R (hunk 1 above)
Comment (ProfessorP-beep) [reply R/derive_extreme_event.R]: Changes made and pushed. Its going through the checks now.

File: R/derive_extreme_event.R [hunk 2]
```diff
@@ -96,7 +96,10 @@
 #'       the number of the event.
 #'       1. Only the variables specified for the `keep_source_vars` field of the
 #'       event, and the by variables (`by_vars`) and the variables created by
-#'       `set_values_to` are kept.
+#'       `set_values_to` are kept. If `keep_source_vars = NULL` is used for an event
+#'       in `derive_extreme_event()` the value of the `keep_source_vars` argument of
+#'       `derive_extreme_event()` is used. In this case the `keep_source_vars` field of
+#'       the event will take precedence over the value of the `keep_source_vars` argument.
```
Comment (bundfussr) [inline R/derive_extreme_event.R:102]: Sorry, I think my QC comment was not clear.

I meant that we should add a note regarding precedence in the description of the `keep_source_vars` argument (line 72-74).
I think we do not need "In this case the `keep_source_vars` field of the event will take precedence over the value of the `keep_source_vars` argument." here.

**Review by bundfussr — APPROVED** [review]

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # admiral (development version)
 
+## Documentation
+
+- Documentation for `derive_extreme_event()` has been updated to include a description for the value of `derive_extreme_event()` when `keep_source_vars = NULL`
```
Comment (bms63) [inline NEWS.md:5]: Hi @ProfessorP-beep - can you move this into the already existing Documentation section and place the issue number at the end.  pkgdown makes it into a dynamic link that can take the user to github!

**Comment by bms63** [comment]
Thank you for fixing that one!!  We have some more challenging issues if you would like to jump in the deep end.   I can try and pick one out.  If not, I think the low hanging fruit is already picked, but new issues come in pretty regularly.

**Comment by ProfessorP-beep** [comment]
Yea, sure! If there is one you suggest I can hop on or take a look at the open list.

**Pull request merged**